crawler/processing/create_plate_exporter.py

```diff
@@ -135,4 +175,146 @@ def _record_source_plate_in_mongo_db(self, session: ClientSession) -> ExportResu
+                        )
+                    )
+
+                return ExportResult(success=False, create_plate_errors=create_plate_errors)
+        except Exception as ex:
+            LOGGER.critical(f"Error accessing MongoDB during export of samples for message UUID '{message_uuid}': {ex}")
+            LOGGER.exception(ex)
+
+            raise TransientRabbitError(
+                f"There was an error updating MongoDB while exporting samples for message UUID '{message_uuid}'."
+            )
+
+        self._samples_inserted = len(result.inserted_ids)
+        LOGGER.info(f"{self._samples_inserted} samples inserted into mongo.")
+
+        return ExportResult(success=True, create_plate_errors=[])
+
+    def _map_sample_to_mongo(self, sample, index):
+        return {
+            FIELD_MONGO_DATE_TESTED: sample.tested_date.value,
+            FIELD_MONGO_LAB_ID: self._message.lab_id.value,
+            FIELD_MONGO_RNA_ID: sample.rna_id.value,
+            FIELD_MONGO_ROOT_SAMPLE_ID: sample.root_sample_id.value,
+            FIELD_MONGO_COG_UK_ID: sample.cog_uk_id.value,
+            FIELD_MONGO_RESULT: sample.result.value.capitalize(),
+            FIELD_SOURCE: self._message.centre_config[CENTRE_KEY_NAME],
+            FIELD_PLATE_BARCODE: self._message.plate_barcode.value,
+            FIELD_COORDINATE: normalise_plate_coordinate(sample.plate_coordinate.value),
+            FIELD_MONGO_SAMPLE_INDEX: index + 1,
+            FIELD_MONGO_MESSAGE_UUID: self._message.message_uuid.value,
+            FIELD_MONGO_FILTERED_POSITIVE: sample.fit_to_pick.value,
+            FIELD_MUST_SEQUENCE: sample.must_sequence.value,
+            FIELD_PREFERENTIALLY_SEQUENCE: sample.preferentially_sequence.value,
+            FIELD_LH_SAMPLE_UUID: sample.sample_uuid.value,
+            FIELD_LH_SOURCE_PLATE_UUID: self._plate_uuid,
+            FIELD_CREATED_AT: datetime.utcnow(),
+            FIELD_UPDATED_AT: datetime.utcnow(),
+        }
+
+    def _record_samples_in_dart(self):
+        LOGGER.info("Adding to DART")
+        message_uuid = self._message.message_uuid.value
+        plate_barcode = self._message.plate_barcode.value
+
+        if (sql_server_connection := create_dart_sql_server_conn(self._config)) is None:
+            error_description = (
+                f"Error connecting to DART database for plate with barcode '{plate_barcode}' "
+                f"in message with UUID '{message_uuid}'"
+            )
+            LOGGER.critical(error_description)
+
+            return ExportResult(
+                success=False,
+                create_plate_errors=[
+                    CreatePlateError(
+                        type=ErrorType.ExportingPostFeedback,  # This error will only reach the imports record
+                        origin=RABBITMQ_CREATE_FEEDBACK_ORIGIN_ROOT,
+                        description=error_description,
+                    )
+                ],
+            )
```

Comment:
I would move this to another separate method, seems the error connection behaviour, so if someone is interested in the detail can go to that method to check how it was implemented